display_list/display_list_builder.cc

```diff
@@ -77,36 +80,56 @@ DisplayListBuilder::~DisplayListBuilder() {
 
 void DisplayListBuilder::onSetAntiAlias(bool aa) {
   Push<SetAntiAliasOp>(0, 0, current_anti_alias_ = aa);
+  if (complexity_accumulator_)
```

Comment:
Will it ever be null? (Also, do native style guidelines require parentheses?)
We will probably always have a backup implementation?

Reply:
This shouldn't be null. Maybe it'd be best to roll the `DisplayListNaiveComplexityCalculator` into the `DisplayListComplexityCalculator` base class so that it's no longer abstract then any subclasses would implement more reasonable heuristics for a given backend.

I'll remove the `if` statements at a minimum, though.